cpp/src/gandiva/tests/hash_test.cc

```diff
@@ -147,4 +153,252 @@ TEST_F(TestHash, TestBuf) {
+
+    auto node_d = TreeExprBuilder::MakeField(field_d);
+    auto hashSha128_4 = TreeExprBuilder::MakeFunction("hashSHA128",
+													  {node_d}, utf8());
+    auto expr_3 = TreeExprBuilder::MakeExpression(hashSha128_4, res_3);
+
+    // Build a projector for the expressions.
+    std::shared_ptr<Projector> projector;
+    auto status =
+            Projector::Make(schema, {expr_0, expr_1, expr_2, expr_3},
+							TestConfiguration(), &projector);
+    EXPECT_TRUE(status.ok()) << status.message();
+
+    // Create a row-batch with some sample data
+    int num_records = 2;
+    auto validity_array = {false, true};
+
+    auto array_int32 =
+            MakeArrowArrayInt32({1, 0}, validity_array);
+
+    auto array_int64 =
+            MakeArrowArrayInt64({1, 0}, validity_array);
+
+    auto array_float32 =
+            MakeArrowArrayFloat32({1.0, 0.0}, validity_array);
+
+    auto array_float64 =
+            MakeArrowArrayFloat64({1.0, 0.0}, validity_array);
+
+    // prepare input record batch
+    auto in_batch = arrow::RecordBatch::Make(schema, num_records,
+											 {array_int32, array_int64,
+			 array_float32, array_float64});
+
+    // Evaluate expression
+    arrow::ArrayVector outputs;
+    status = projector->Evaluate(*in_batch, pool_, &outputs);
+    EXPECT_TRUE(status.ok());
+
+    EXPECT_ARROW_ARRAY_EQUALS(outputs.at(0), outputs.at(1));
+    EXPECT_ARROW_ARRAY_EQUALS(outputs.at(1), outputs.at(2));
+    EXPECT_ARROW_ARRAY_EQUALS(outputs.at(2), outputs.at(3));
+}
+
+TEST_F(TestHash, TestSha128Varlen) {
+    // schema for input fields
+    auto field_a = field("a", utf8());
+    auto schema = arrow::schema({field_a});
+
+    // output fields
+    auto res_0 = field("res0", utf8());
+
+    // build expressions.
+    // hashSHA128(a)
+    auto node_a = TreeExprBuilder::MakeField(field_a);
+    auto hashSha128 = TreeExprBuilder::MakeFunction("hashSHA128",
+													{node_a}, utf8());
+    auto expr_0 = TreeExprBuilder::MakeExpression(hashSha128, res_0);
+
+    // Build a projector for the expressions.
+    std::shared_ptr<Projector> projector;
+    auto status =
+            Projector::Make(schema, {expr_0}, TestConfiguration(), &projector);
+    EXPECT_TRUE(status.ok()) << status.message();
+
+    // Create a row-batch with some sample data
+    int num_records = 3;
+
+    std::string first_string = "ði ıntəˈnæʃənəl fəˈnɛtık əsoʊsiˈeıʃn\nY [ˈʏpsilɔn], "
+							   "Yen [jɛn], Yoga [ˈjoːgɑ]";
+    std::string second_string = "ði ıntəˈnæʃənəl fəˈnɛtık əsoʊsiˈeın\nY [ˈʏpsilɔn], "
+								"Yen [jɛn], Yoga [ˈjoːgɑ] コンニチハ";
+
+    auto array_a =
+            MakeArrowArrayUtf8({"foo", first_string, second_string},
+							   {false, true, true});
+
+    // prepare input record batch
+    auto in_batch = arrow::RecordBatch::Make(schema, num_records, {array_a});
+
+    // Evaluate expression
+    arrow::ArrayVector outputs;
+    status = projector->Evaluate(*in_batch, pool_, &outputs);
+    EXPECT_TRUE(status.ok());
```

Comment:
I believe in Arrow we have ASSERT_OK, which should porbably be used (I don't think you want to continue if status isn't OK?

Reply:
Changed the EXPECT_TRUE to ASSERT_OK